presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2115,6 +2135,24 @@ private DynamicFilterSourceOperatorFactory createDynamicFilterSourceOperatorFact
                     });
         }
 
+        private Optional<LocalDynamicFilter> createDynamicFilter(SemiJoinNode node, LocalExecutionPlanContext context)
+        {
+            if (!isEnableDynamicFiltering(context.getSession()) || !node.getDynamicFilterId().isPresent()) {
+                return Optional.empty();
+            }
+
+            String dynamicFilterId = node.getDynamicFilterId().get();
+            Multimap<String, Symbol> probeSymbols = ImmutableMultimap.of(dynamicFilterId, node.getSourceJoinSymbol());
+            Map<String, Integer> buildChannels = ImmutableMap.of(dynamicFilterId, node.getFilteringSource().getOutputSymbols().indexOf(node.getFilteringSourceJoinSymbol()));
+
+            LocalDynamicFilter localDynamicFilter = new LocalDynamicFilter(probeSymbols, buildChannels, 1);
```

Comment:
I think you should implement or extend similar method as `io.prestosql.sql.planner.LocalDynamicFilter#create`. Note that for joins we don't create dynamic filter operator if table scan is in different stage as join.

Reply:
Right, thanks for the correction